Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3874,6 +3877,95 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
         wantsWheelEventsChanged();
 }
 
+#if PLATFORM(IOS_FAMILY)
+
+void UnifiedPDFPlugin::setSelectionRange(FloatPoint pointInRootView, TextGranularity granularity)
+{
+    RetainPtr selection = [&] -> PDFSelection * {
+        auto pointInPlugin = convertFromRootViewToPlugin(roundedIntPoint(pointInRootView));
+        auto pointInDocument = convertDown<FloatPoint>(CoordinateSpace::Plugin, CoordinateSpace::PDFDocumentLayout, pointInPlugin);
+        auto pageIndex = m_presentationController->nearestPageIndexForDocumentPoint(pointInDocument);
+        auto pointInPage = convertDown(CoordinateSpace::PDFDocumentLayout, CoordinateSpace::PDFPage, pointInDocument, pageIndex);
+        RetainPtr page = m_documentLayout.pageAtIndex(pageIndex);
+        switch (granularity) {
+        case TextGranularity::WordGranularity:
+            return [page selectionForWordAtPoint:pointInPage];
+        case TextGranularity::LineGranularity:
+            return [page selectionForLineAtPoint:pointInPage];
```

Comment:
I think we can add CharacterGranularity support too, do you mind adding that or leaving a FIXME so we can evaluate if that even matters?

```suggestion
        case TextGranularity::CharacterGranularity:
            return [m_pdfDocument selectionFromPage:page.get() atPoint:pointInPage toPage:page.get() atPoint:pointInPage withGranularity:PDFSelectionGranularityCharacter];
        case TextGranularity::WordGranularity:
            return [page selectionForWordAtPoint:pointInPage];
        case TextGranularity::LineGranularity:
            return [page selectionForLineAtPoint:pointInPage];
```

Reply:
I wonder if `selectionFor[Word|Line]AtPoint:` behaves differently from `selectionFromPage:atPoint:toPage:atPoint:withGranularity:PDFSelectionGranularity[Word|Line]`

Reply:
Oh, interesting — I can try using `-selectionFromPage:atPoint:toPage:atPoint:withGranularity:` here...